core/controller/src/main/scala/whisk/core/controller/Triggers.scala

```diff
@@ -122,72 +126,135 @@ trait WhiskTriggersApi extends WhiskCollectionAPI {
-                        Unmarshal(response.entity).to[JsObject].map { a =>
-                          logging.info(this, s"${rule.action} activated ${a.fields("activationId")}")
-                        }
-                      case NotFound =>
-                        response.discardEntityBytes()
-                        logging.info(this, s"${rule.action} failed, action not found")
-                      case _ =>
-                        Unmarshal(response.entity).to[String].map { error =>
-                          logging.warn(this, s"${rule.action} failed due to $error")
+          // List of active rules associated with the trigger
+          val activeRules = trigger.rules map { _.filter((r) => r._2.status == Status.ACTIVE) } getOrElse Map.empty
+
+          if (activeRules.nonEmpty) {
+            val args = trigger.parameters.merge(payload)
+
+            // Iterate through each active rule; invoking each mapped action
+            val actionLogList = activeRules
+              .map {
+                case (ruleName, rule) =>
+                  // Build the url to invoke an action mapped to the rule
+                  val actionNamespace = rule.action.path.root.asString
+                  val actionPath = {
+                    rule.action.path.relativePath.map { pkg =>
+                      (Path.SingleSlash + pkg.namespace) / rule.action.name.asString
+                    } getOrElse {
+                      Path.SingleSlash + rule.action.name.asString
+                    }
+                  }.toString
+
+                  val actionUrl = Path("/api/v1") / "namespaces" / actionNamespace / "actions"
+                  val request = HttpRequest(
+                    method = POST,
+                    uri = url.withPath(actionUrl + actionPath),
+                    headers =
+                      List(Authorization(BasicHttpCredentials(user.authkey.uuid.asString, user.authkey.key.asString))),
+                    entity = HttpEntity(MediaTypes.`application/json`, args.getOrElse(JsObject()).compactPrint))
+
+                  // Invoke the action. Retain action results for inclusion in the trigger activation record
+                  Http()
+                    .singleRequest(request)
+                    .flatMap {
+                      response =>
+                        response.status match {
+                          case OK | Accepted =>
+                            Unmarshal(response.entity).to[JsObject].map { a =>
+                              logging.info(
+                                this,
+                                s"trigger-fired rule '${rule.action}' invoked with activation ${a.fields("activationId")}")
+                              (ruleName.asString -> triggerActivationLogMsg(
+                                entityName.asString,
+                                ruleName.asString,
+                                rule.action.asString,
+                                InfoLevel,
```

Comment:
what is the log level needed for?

Reply:
i was thinking it would be good to flag invocation failures with a tag that can eventually be consumed by a logging service.  here's some sample output
```
  "logs": [
      "[2018-01-16T16:23:17.174Z] [INFO] [guest/wtrig] [guest/wrule] [guest/weather] 'guest/weather' activated \"3c2aaf37e07549cbaaaf37e07519cb19\"",
      "[2018-01-16T16:23:17.175Z] [INFO] [guest/wtrig] [guest/wrule2] [guest/echo-env-web] 'guest/echo-env-web' activated \"8e2b7c0f04084c53ab7c0f04087c5386\"",
      "[2018-01-16T16:23:16.948Z] [ERROR] [guest/wtrig] [guest/wrule3] [guest/badrule] 'guest/badrule' failed, action not found"
  ],
```

Reply:
thanks for the clarification.
we'll want this to be something easily consumable; @starpit what do think?
we can use a fixed format like this, or perhaps a json representation logged to a single line.

Reply:
trigger activations will still have activation records, but rules will not? and if, e.g. in this case, the rule's action is not found, there will be a log entry in that trigger fire's activation record?  or are these separate log records?

Reply:
with this pr, only two kinds of activation records will be created - one for the trigger and one for each associated action.  the rule activation will no longer be created.  the trigger activation log entries shown above are meant to assist in tracking down the action results that were generated in response to the trigger event.

Reply:
thanks for the help in my understanding. at first glance, this seems like information better suited for a statusCode?

Reply:
Yes. But here are some caveats. We can’t use the trigger activation result because it’s already used. It records the triggers incoming payload. 

There are multiple results, one for each action that’s activated. 

I do think we can use the activation response schema to record the result of its action that’s activated. And store these in an array. The question is do we log these as log lines as in the current form, use annotations or use a single log line that is formatted as a json string.

Reply:
"single log line ... as a json string". is the suggestion that we change the entire log record to a JsonFormat structure, or that we have a hybrid, with "what we have now" using the current positional encoding, and then subrecords formatted as JSON?

there is some nascent desire for allowing the actions themselves to communicate structured information to the user; e.g. internal timing information. so this is a somewhat broader topic.

Reply:
I'm strictly speaking about the trigger activation, not action logs.
The schema doesn't constrain log lines. It's a string. That we generate a `[timestamp] [stream] text` format today is a convenience generated by the invoker.

I'm suggesting that the timestamp and stream are meaningless here. So it doesn't make sense to follow these conventions. Instead, a single JSON string of the action post results is more appropriate in my view.

How logs generate their log lines, and how those are parsed, indexed, and enriched is a separate issue entirely in my opinion.

Reply:
@csantanapr and i talked about this format a bit and we liked the idea of maintaining the log format of existing action invocation logs - iff we want to use the activation records `logs` field.  potentially a logging service, like logmet, could consume these and make trigger/action search/correlation easier.

Reply:
just to wrap up this topic.  @csantanapr, @rabbah and i spoke and decided to write out an array of json strings instead of the above log entries.  each json string would be a single json object representing the status/result of the async rule/action activation.  for example
```d
    "logs": [
        "{\"statusCode\":0,\"success\":true,\"activationId\":\"ae2d98c6b49b4a32ad98c6b49bba32c5\",\"rule\":\"guest/wrule\",\"action\":\"guest/echo-env-web\"}",
        "{\"statusCode\":0,\"success\":true,\"activationId\":\"32e3afff8a7f4c31a3afff8a7fbc319b\",\"rule\":\"guest/wrule2\",\"action\":\"guest/echo-env-web\"}",
        "{\"statusCode\":1,\"success\":false,\"rule\":\"guest/wrule3\",\"error\":\"The requested resource does not exist.\",\"action\":\"guest/nonexistent-action\"}"
    ]
```